subsys/net/lib/lwm2m/lwm2m_senml.patch

```diff
@@ -0,0 +1,335 @@
+ 						  (&(*result)._value_int))) &&
+-			      ((((*result)._value_int >=
+-				 -9223372036854775807LL) &&
+-				((*result)._value_int <=
+-				 9223372036854775807LL)) ||
++			      ((((*result)._value_int >= INT64_MIN) &&
++				((*result)._value_int <= INT64_MAX)) ||
+ 			       (zcbor_error(state, ZCBOR_ERR_WRONG_RANGE),
+ 				false))) &&
+ 			     (((*result)._value_choice = _value_int) || 1)) ||
+@@ -194,7 +193,8 @@ static bool decode_record(zcbor_state_t *state, struct record *result)
+ 			     (zcbor_decoder_t *)decode_repeated_record_union,
+ 			     state, (&(*result)._record_union)) &&
+ 		     zcbor_multi_decode(
+-			     0, 3, &(*result)._record__key_value_pair_count,
++			     0, ARRAY_SIZE(result->_record__key_value_pair),
++			     &(*result)._record__key_value_pair_count,
+ 			     (zcbor_decoder_t *)
+ 				     decode_repeated_record__key_value_pair,
+ 			     state, (&(*result)._record__key_value_pair),
+@@ -214,7 +214,7 @@ static bool decode_lwm2m_senml(zcbor_state_t *state, struct lwm2m_senml *result)
+ 
+ 	bool tmp_result =
+ 		(((zcbor_list_start_decode(state) &&
+-		   ((zcbor_multi_decode(1, 99,
++		   ((zcbor_multi_decode(1, ARRAY_SIZE(result->_lwm2m_senml__record),
+ 					&(*result)._lwm2m_senml__record_count,
+ 					(zcbor_decoder_t *)decode_record, state,
+ 					(&(*result)._lwm2m_senml__record),
+diff --git a/subsys/net/lib/lwm2m/lwm2m_senml_cbor_decode.h b/subsys/net/lib/lwm2m/lwm2m_senml_cbor_decode.h
+index 896f8833ba..c12cd50512 100644
+--- a/subsys/net/lib/lwm2m/lwm2m_senml_cbor_decode.h
++++ b/subsys/net/lib/lwm2m/lwm2m_senml_cbor_decode.h
+@@ -1,23 +1,23 @@
++/*
++ * Copyright (c) 2022 Nordic Semiconductor ASA
++ *
++ * SPDX-License-Identifier: Apache-2.0
++ */
+ /*
+  * Generated using zcbor version 0.5.1
+- * https://github.com/NordicSemiconductor/zcbor
++ * https://github.com/zephyrproject-rtos/zcbor
+  * Generated with a --default-max-qty of 99
+  */
+ 
+ #ifndef LWM2M_SENML_CBOR_DECODE_H__
+ #define LWM2M_SENML_CBOR_DECODE_H__
+ 
+-#include "lwm2m_senml_cbor_types.h"
+-#include "zcbor_decode.h"
+-
+ #include <stdbool.h>
+ #include <stddef.h>
+ #include <stdint.h>
+ #include <string.h>
+-
+-#if DEFAULT_MAX_QTY != 99
+-#error "The type file was generated with a different default_max_qty than this file"
+-#endif
++#include "zcbor_decode.h"
++#include "lwm2m_senml_cbor_types.h"
+ 
+ int cbor_decode_lwm2m_senml(const uint8_t *payload, size_t payload_len,
+ 			    struct lwm2m_senml *result,
+diff --git a/subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c b/subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c
+index d2f9bf68f2..5864970b94 100644
+--- a/subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c
++++ b/subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c
+@@ -1,20 +1,21 @@
++/*
++ * Copyright (c) 2022 Nordic Semiconductor ASA
++ *
++ * SPDX-License-Identifier: Apache-2.0
++ */
+ /*
+  * Generated using zcbor version 0.5.1
+- * https://github.com/NordicSemiconductor/zcbor
++ * https://github.com/zephyrproject-rtos/zcbor
```

Comment:
There is no plan to deprecate https://github.com/NordicSemiconductor/zcbor. The plan is that it will continue to be the main development repo, and the zephyrproject-rtos will be a mirror.

Reply:
What I mean here is that the text _https://github.com/NordicSemiconductor/zcbor_ will not be there when the generator recreates the source file.

Reply:
Not sure what you mean, the text "https://github.com/NordicSemiconductor/zcbor" will most likely be there the next time too. But, yes, the patch will likely fail to apply cleanly the next time, but it serves as a record of the changes. Also, it allows you to use git's conflict resolution to fix the conflicts.

Reply:
Hehe, I just rebased onto the newest zephyr main and the patch didn't apply. Turns out the clang-format file was changed in the meantime. Uploaded a resolved version of the patch and files.

Reply:
I've added information in the readme about how to resolve conflicts.